tests/run/for_in_string.pyx

```diff
@@ -224,3 +225,27 @@ def for_pyucs4_in_enumerate_unicode(unicode s):
             return i
     else:
         return 'X'
+
+@cython.test_assert_path_exists("//ForFromStatNode")
+@cython.test_fail_if_path_exists("//ForInStatNode")
+def for_infered_type_in_enumerate_unicode(unicode s):
+    """
+    >>> for_infered_type_in_enumerate_unicode(unicode_abc)
+    ('Py_ssize_t', 'Py_UCS4')
+    'X'
+    >>> for_infered_type_in_enumerate_unicode(unicode_ABC)
+    ('Py_ssize_t', 'Py_UCS4')
+    2
+    >>> for_infered_type_in_enumerate_unicode(unicode_abc_null)
+    ('Py_ssize_t', 'Py_UCS4')
+    'X'
+    >>> for_infered_type_in_enumerate_unicode(unicode_ABC_null)
+    ('Py_ssize_t', 'Py_UCS4')
+    4
+    """
+    print(typeof(i), typeof(c))
```

Comment:
You might want to add a `from __future__ import print_function` at the top of the file.